From 13609c70dac0f54baaf54c492391fc2b5364b352 Mon Sep 17 00:00:00 2001
From: Karthikeyan Periyasamy <quic_periyasa@quicinc.com>
Date: Thu, 16 Mar 2023 11:37:21 +0530
Subject: [PATCH] wifi: ath12k: Skip CPU sync for REO in unmap DMA memory

DMA unmap sync take more cpu cycle compare to ARM asm sync operation.
So avoid dma_unmap with sync and added ARM asm sync before dma_unmap
in REO processing.

Signed-off-by: Karthikeyan Periyasamy <quic_periyasa@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/dp_rx.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/dp_rx.c b/drivers/net/wireless/ath/ath12k/dp_rx.c
index acf284d58338..05501b0f03f5 100644
--- a/drivers/net/wireless/ath/ath12k/dp_rx.c
+++ b/drivers/net/wireless/ath/ath12k/dp_rx.c
@@ -8,6 +8,7 @@
 #include <linux/kernel.h>
 #include <linux/skbuff.h>
 #include <crypto/hash.h>
+#include <asm/cacheflush.h>
 #include "core.h"
 #include "debug.h"
 #include "debugfs_htt_stats.h"
@@ -3331,9 +3332,10 @@ try_again:
 		spin_unlock_bh(&src_ab->dp.rx_desc_lock);
 
 		rxcb = ATH12K_SKB_RXCB(msdu);
-		dma_unmap_single(src_ab->dev, rxcb->paddr,
-				 msdu->len + skb_tailroom(msdu),
-				 DMA_FROM_DEVICE);
+		dmac_inv_range_no_dsb(msdu->data, msdu->data + (msdu->len + skb_tailroom(msdu)));
+		dma_unmap_single_attrs(src_ab->dev, rxcb->paddr,
+				       msdu->len + skb_tailroom(msdu),
+				       DMA_FROM_DEVICE, DMA_ATTR_SKIP_CPU_SYNC);
 
 		num_buffs_reaped[src_ab->chip_id]++;
 
-- 
2.37.0

